Automattic/wp-calypso
# Business & Merchant Survey: Add the survey modal

Related to pekYwv-4f0-p2#comment-3759

## Proposed Changes

This PR shows the following survey modal in My Home:

<img width="938" alt="image" src="https://github.com/user-attachments/assets/2d253c1a-8640-4e59-9eb4-c756d294ddcd">

for sites with the following criteria:

- is on any Business or Commerce plans,
- purchased the plan on or after 1 June 2024,
- owned the plan for at least 7 days.

The link to the survey depends on the launch status:

- Launched site: https://automattic.survey.fm/businesses-merchants-who-have-launched-website
- Unlaunched site: https://automattic.survey.fm/businesses-merchants-who-not-launched-website

## Why are these changes being made?

We want to collect feedback from such users about launching (or not launching) their sites.

## Testing Instructions

1. Open My Home of a recent Free test site of yours (created at most 7 days ago).
    - Verify that you don't see the modal.
1. Open My Home of an old Business/Commerce test site of yours (created at most 1 June 2024).
    - If you don't have such sites, you can test by checking out this PR locally and manipulate the date threshold.
    - Verify that you don't see the modal.
1. Open My Home of recent Business/Commerce test site of yours (created after 1 June 2024 and  at most 7 days ago).
    - If you don't have such sites, you can test by checking out this PR locally and manipulate the date threshold.
    - Verify that you SEE the modal.
1. Launch the site.
    - Verify that the link is correct as explained above (launched).
1. Set the site as Coming Soon
    - Verify that the link is correct as explained above (unlaunched).

## Pre-merge Checklist
